openedx/core/djangoapps/content/course_overviews/tests.py

```diff
@@ -485,3 +492,266 @@ def test_get_all_courses_by_org(self):
+        self.assertTrue(hasattr(course_overview_after, 'image_set'))
+        image_set = course_overview_after.image_set
+        self.assertEqual(image_set.small_url, broken_small_url)
+        self.assertEqual(image_set.large_url, broken_large_url)
+
+        # But because we've disabled it, asking for image_urls should give us
+        # the raw source image for all resolutions, and not our broken images.
+        expected_url = course_image_url(course)
+        self.assertEqual(
+            course_overview_after.image_urls,
+            {
+                'raw': expected_url,
+                'small': expected_url,
+                'large': expected_url
+            }
+        )
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_error_generating_thumbnails(self, modulestore_type):
+        """
+        Test a scenario where thumbnails cannot be generated.
+
+        We need to make sure that:
+
+        a) We don't cause any 500s to leak out. A failure to generate thumbnails
+           should never cause CourseOverview generation to fail.
+        b) We return the raw course image for all resolutions.
+        c) We don't kill our CPU by trying over and over again.
+        """
+        with mock.patch('openedx.core.lib.courses.create_course_image_thumbnail') as patched_create_thumbnail:
+            # Strictly speaking, this would fail anyway because there's no data
+            # backing sample_image.png, but we're going to make the side-effect
+            # more dramatic. ;-)
+            fake_course_image = 'sample_image.png'
+            patched_create_thumbnail.side_effect = Exception("Kaboom!")
+
+            # This will generate a CourseOverview and verify that we get the
+            # source image back for all resolutions.
+            course_overview = self._assert_image_url_values(modulestore_type, fake_course_image)
+
+            # Make sure we were called (i.e. we tried to create the thumbnail)
+            patched_create_thumbnail.assert_called()
+
+        # Now an image set does exist, even though it only has blank values for
+        # the small and large urls.
+        self.assertTrue(hasattr(course_overview, 'image_set'))
+        self.assertEqual(course_overview.image_set.small_url, '')
+        self.assertEqual(course_overview.image_set.large_url, '')
+
+        # The next time we create a CourseOverview, the images are explicitly
+        # *not* regenerated.
+        with mock.patch('openedx.core.lib.courses.create_course_image_thumbnail') as patched_create_thumbnail:
+            course_overview = CourseOverview.get_from_id(course_overview.id)
+            patched_create_thumbnail.assert_not_called()
+
+    @ddt.data(
+        *itertools.product(
+            [ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split],
+            [True, False]
+        )
+    )
+    @ddt.unpack
+    def test_happy_path(self, modulestore_type, create_after_overview):
```

Comment:
Consider testing with different image sizes set in the config and different image sizes of the raw image.